src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -0,0 +1,202 @@
+
+        return network_id_table
+
+    def _get_active_connection(self, network_id):
+        """Returns the active connection object associated with the given network id."""
+
+        # The active connection objects returned by NetworkManager are very short-lived and
+        #   not directly accessible from the connection object.
+
+        active_connection_list = NetworkManager.NetworkManager.ActiveConnections
+        self._run_proxy_call(active_connection_list)
+
+        for listed_active_connection in active_connection_list:
+            if listed_active_connection.Id == network_id:
+                self.logger.trace('Found active connection.')
+                return listed_active_connection
+
+        return None
+
+    def _get_connection_state(self, network_id):
+        """Returns the state of the connection with the given network id."""
+        state = NM_CONNECTION_DISCONNECTED
+
+        active_connection = self._get_active_connection(network_id)
+
+        if active_connection is None:
+            self.logger.warning('Connection disconnected.')
+
+        else:
+            if hasattr(active_connection, 'State'):
+                if active_connection.State is NetworkManager.NM_ACTIVE_CONNECTION_STATE_ACTIVATING:
+                    state = NM_CONNECTION_ACTIVATING
+                if active_connection.State is NetworkManager.NM_ACTIVE_CONNECTION_STATE_ACTIVATED:
+                    state = NM_CONNECTION_ACTIVE
+
+        return state
+
+    def _get_device_objects(self, wired_interface_name, wifi_interface_name):
+        """Store references to the device objects that are used frequently."""
+
+        for device in NetworkManager.NetworkManager.GetDevices():
+            if device.Interface == wired_interface_name:
+                self.wired_device = device
+
+            if device.Interface == wifi_interface_name:
+                self.wireless_device = device
+
+        if self.wired_device is None:
+            raise DeviceNotFoundException('Defined wired device %s was not found.' %
+                wired_interface_name)
+
+        if self.wireless_device is None:
+            raise DeviceNotFoundException('Defined wireless device %s was not found.' %
+                wifi_interface_name)
+
+    def _get_device_for_connection(self, connection):
+        """Get the device object a connection object needs to connect with."""
+
+        network_device = self.wireless_device
+
+        connection_id = connection.GetSettings()['connection']['id']
+        if connection_id == self.wired_network_name:
+            network_device = self.wired_device
+
+        return network_device
+
+    def _wait_for_connection(self, connection):
+        """Wait timeout number of seconds for an active connection to be ready.
+        return True if it connects within timeout, False otherwise.
+        """
+        success = False
+        give_up = False
+        time_to_give_up = time.time() + self.network_activation_timeout
```

Comment:
I'd probably assign this variable immediately before the loop for chrono accuracy.